# Holberton School - 0x05. Python - Exceptions
Description

The focus of this project is to learn:
* when using exceptions are neccessary and how to handle them
* the difference between errors and exceptions
* the purpose of catching exceptions
* how to raisea builtin expression
* when implementing a clean-up action is neccessary after catching an exception

## Environment
* __Environment:__ Ubuntu 14.04 LTS
* __python3 version:__ 3.4.3
* __style:__ PEP 8

## New commands / functions used:
TODO

## Helpful Links
* https://docs.python.org/3.4/tutorial/errors.html
* https://www.youtube.com/watch?v=7vbgD-3s-w4
* https://docs.python.org/3.4/tutorial/index.html

## File Descriptions
- `0-safe_print_list.py`: prints x elements of a list
- `1-safe_print_integer.py`: prints an integer with "{:d}".format()
- `2-safe_print_list_integers.py`: prints the first x elemnts of a list that are integers
- `3-safe_print_division.py`: divides two integers and prints the result
- `4-list_division.py`: divides element by element between 2 lists
- `5-raise_exception.py`: raises a type exception
- `6-raise_exception_msg.py`: raises a name exception with a message
- `100-safe_print_integer_err.py`: prints an integer
- `101-safe_function.py`: executes a function safely

## Author
Stuart Kuredjian

twitter: @stueygk

linkdin: https://www.linkedin.com/in/stuart-kuredjian-8100a273/

## License
Public Domain, no copyright protection
